node/state/state.go

```diff
@@ -17,32 +17,36 @@ import (
 
 var log = logging.Logger("state-access-constructor")
 
-// Components provides all components necessary to construct the
+// Module provides all components necessary to construct the
 // state service.
-func Components(coreCfg core.Config, keyCfg key.Config) fx.Option {
-	return fx.Options(
+func Module(coreCfg core.Config, keyCfg key.Config) fx.Option {
+	return fx.Module(
+		"state",
 		fx.Provide(Keyring(keyCfg)),
-		fx.Provide(CoreAccessor(coreCfg.IP, coreCfg.RPCPort, coreCfg.GRPCPort)),
-		fx.Provide(Service),
+		fx.Provide(fx.Annotate(CoreAccessor(coreCfg.IP, coreCfg.RPCPort, coreCfg.GRPCPort),
+			fx.OnStart(func(ctx context.Context, accessor state.Accessor) error {
+				return accessor.Start(ctx)
+			}),
+			fx.OnStop(func(ctx context.Context, accessor state.Accessor) error {
+				return accessor.Stop(ctx)
+			}),
+		)),
+		fx.Provide(fx.Annotate(Service,
```

Comment:
You can use here the constructor directly from the state pkg. that's the whole point of using this new feature.

Reply:
https://github.com/celestiaorg/celestia-node/pull/997/commits/2e40ae048785bf53e9bc3701d9e361236389f34e